### ISSUE TITLE
Unable to relay with multiple recipients

### ISSUE THREAD

**Issue by BrianInAz** [rnwood/smtp4dev#462]
re: #455 

I have two addresses whitelisted for relay:

```
//Specifies a list of recipient addresses for which messages will be relayed. An empty list means that no messages are relayed.
//Default value: []
     "AllowedEmails": [
 		"test1@test.com",
 		"test2@test.com"
],
```

When I try to relay to another SMTP server with a **single** to: address (test1@test.com **OR** test2@test.com) it works just fine. BUT, if I am trying to relay a message that has **two** recipients, I get an error.

Test Output:

```
smtp4dev version 3.1.1
https://github.com/rnwood/smtp4dev

.NET Core runtime version: Rnwood.Smtp4dev-win-x64-3.1.1


 > For help use argument --help

Using Sqlite database at C:\Users\b\AppData\Roaming\smtp4dev\database.db

TLS mode: None
SSL/TLS is now disabled by default
To enable use set TlsMode option (values 'ImplicitTls' or 'StartTls') on command line or in appsettings.json and follow the instruction about hostname and certificate trust on first startup.

SMTP Server is listening on port 25.
Keeping last 100 messages and 100 sessions.
Hosting environment: Production
Content root path: C:\Rnwood.Smtp4dev-win-x64-3.1.1
Now listening on: http://localhost:5000
Application started. Press Ctrl+C to shut down.
Session started. Client address 127.0.0.1.
Message received. Client address 127.0.0.1. From testFrom@test.com. To test1@test.com, test2@test.com.
Processing received message
Relaying message to test1@test.com, test2@test.com
Can not relay message to test1@test.com, test2@test.com: MimeKit.ParseException: Unexpected token at offset 14
   at MimeKit.MailboxAddress.Parse(ParserOptions options, String text)
   at Rnwood.Smtp4dev.Server.Smtp4devServer.RelayMessage(Message message) in /home/vsts/work/1/s/Rnwood.Smtp4dev/Server/Smtp4devServer.cs:line 324
Processing received message DONE
Session completed. Client address 127.0.0.1. Number of messages 1.
Session started. Client address 127.0.0.1.
Message received. Client address 127.0.0.1. From testFrom@test.com. To test1@test.com.
Processing received message
Relaying message to test1@test.com
Processing received message DONE
Session completed. Client address 127.0.0.1. Number of messages 1.
Session started. Client address 127.0.0.1.
Message received. Client address 127.0.0.1. From testFrom@test.com. To test2@test.com.
Processing received message
Relaying message to test2@test.com
Processing received message DONE
Session completed. Client address 127.0.0.1. Number of messages 1.
```

I looked at `/Rnwood.Smtp4dev/Server/Smtp4devServer.cs:line 324` and found the logic used to handle multiple addresses is looking for a semicolon to split on:
```
foreach (string envelopeTo in message.To.Split(";"))
```

I went and found the code calling RelayMessage() and it is passing through multiple addresses split by comma instead of semicolon.

```
        private async Task OnMessageReceived(object sender, MessageEventArgs e)
        {
            using (Stream stream = e.Message.GetData().Result)
            {
                string to = string.Join(", ", e.Message.Recipients);
                Console.WriteLine($"Message received. Client address {e.Message.Session.ClientAddress}. From {e.Message.From}. To {to}.");
                Message message = new MessageConverter().ConvertAsync(stream, e.Message.From, to).Result;
                message.IsUnread = true;


                await QueueTask(() =>
                {
                    Console.WriteLine("Processing received message");
                    Smtp4devDbContext dbContext = dbContextFactory();

                    RelayMessage(message);
```

So I think this could be fixed by changing line 324 to split on comma:
```
foreach (string envelopeTo in message.To.Split(","))
```

Thanks! Love this app.

**Comment by rnwood** [comment]
Thanks for reporting. I'll push out a minor release with this fix soon.
I will address the root cause (bad API design - this property should be an array) longer term.

**Comment by rnwood** [comment]
Should be resolved in 3.1.1.1 and latest prerelease versions.
